network.go

```diff
@@ -1886,9 +1891,19 @@ func (n *network) ResolveIP(ip string) string {
 
 	nwName := n.Name()
 
-	ipInfo, ok := sr.ipMap[ip]
-
-	if !ok || ipInfo.extResolver {
+	elemSet, ok := sr.ipMap.Get(ip)
+	if !ok || len(elemSet) == 0 {
+		return ""
+	}
+	// NOTE it is possible to have more than one element in the Set, this will happen
+	// because of interleave of diffent events from differnt sources (local container create vs
+	// network db notifications)
+	// In such cases the resolution will be based on the first element of the set, and can vary
+	// during the system stabilitation
+	ipInfo, ok := elemSet[0].(ipInfo)
+	if !ok {
+		setStr, b := sr.ipMap.String(ip)
+		logrus.Warnf("Something is wrong with the ipInfo set for key %s set:%t %s", ip, b, setStr)
```

Comment:
suggestion? It will be !ok if the element inserted is not of type ipInfo that is strange and major bug